### DISCUSSION TITLE
Python: error: cannot intersect: ~2.6 && ^3.7

### DISCUSSION THREAD

**Discussion by nickali** [pkgxdev/pkgx#280] [Help] [answered]
I'm putting this here instead of creating an issue because I think this is really a user issue than tea.

I'm running macOS Monterey 12.6.2 which has python 3 but I'd like use python 2.6 for a project. So I'm trying to get 2.6 set up in a completely empty directory:

```sh
2:03:37 ❯ tea +python.org^2.6.0 python
tea: resolving package graph
error: cannot intersect: ^2.6 && ^3.7
```
Alternately, I've tried the following:

```sh
2:07:14 ❯ tea -X +python.org^2.6.0 python
tea: resolving package graph

panic: split tea. we’re sorry and we’ll fix it… but you have to report the bug!

    https://github.com/teaxyz/cli/issues/new?title=panic%3Acannot+intersect%3A+%5E2.6+%26%26+%5E3.7

----------------------------------------------------->> attachment begin
Error: cannot intersect: ^2.6 && ^3.7
    at Module.intersect (file:///opt/tea.xyz/src/v0.16.4/src/utils/semver.ts:273:32)
    at ascend (file:///opt/tea.xyz/src/v0.16.4/src/prefab/hydrate.ts:47:55)
    at async go (file:///opt/tea.xyz/src/v0.16.4/src/prefab/hydrate.ts:59:9)
    at async hydrate (file:///opt/tea.xyz/src/v0.16.4/src/prefab/hydrate.ts:67:9)
    at async install (file:///opt/tea.xyz/src/v0.16.4/src/app.exec.ts:443:28)
    at async prepare_exec_cmd (file:///opt/tea.xyz/src/v0.16.4/src/app.exec.ts:217:22)
    at async X (file:///opt/tea.xyz/src/v0.16.4/src/app.X.ts:17:29)
    at async file:///opt/tea.xyz/src/v0.16.4/src/app.ts:35:9
<<------------------------------------------------------- attachment end
```
What am I doing wrong?

**Comment by mxcl** [comment, marked as answer]
Thanks for the report! This is two tea bugs due to our relative immaturity:

1. We don’t package python 2 (but we should)
2. Our pip package requires python >=3.7 because we don’t yet support varying the version of a dependency by the version of that package.

> 2. happens because we bundle pip separately to python which is our choice so that the user can have complete control over their entire dependency stack, but then pip itself varies its python requirements over the years.

Anyway we'll try to get these fixed thanks.

**Reply by mxcl** [reply]
it occurs that this happens because we are trying to fetch pip magically (it's listed as a “companion pkg”). You can thus work around this with `--disable-magic` or `-m`. Though that only gets you to the next error: no python 2. But I'm trying to package that RN.

**Reply by mxcl** [reply]
k so we cannot build python.org^2 on Darwin aarch64 (maybe with a few diffs it will work, but for now we haven't the bandwidth) which means this package is dependent on us supporting package.yml keys that specify some packages are not bottled for certain platform/arch combinations. 

Refs: https://github.com/teaxyz/pantry.core/commit/c78878a22f4fea933260d2c56205a1cec448286d

**Comment by nickali** [comment]
Yeah, makes sense!
